tests/e2e/scale/test_scale_osds_fill_75%_reboot_workers.py

```diff
@@ -0,0 +1,103 @@
+"""
+This testcase starts with minimum number of osds(one osd)
+on each node and slowly scaling it into 6 osds and then reboot worker
+nodes
+"""
+
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.cluster import count_cluster_osd, validate_osd_utilization
+from ocs_ci.ocs.platform_nodes import AWSNodes
+from ocs_ci.ocs.node import get_typed_nodes, wait_for_nodes_status
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers, tier4
+from ocs_ci.ocs.resources import storage_cluster
+from tests import helpers
+from ocs_ci.utility.utils import ceph_health_check
+
+
+logger = logging.getLogger(__name__)
+
+
+@scale
+@tier4
+@ignore_leftovers
+class TestScaleOSDsRebootNodes(E2ETest):
+    """
+    Add first set of OSD to a minimum cluster with 50%
+    of storage utilization and wait for rebalance
+    Reboot worker nodes after rebalance
+    """
+
+    def test_scale_osds_reboot_nodes(self, pvc_factory):
+        """
+        Check storage utilization, if its less then runs IO,
+        Scale osds from 3-6, check for rebalance and reboot workers
+        """
+        current_osd_count = count_cluster_osd()
+        if current_osd_count == 3:
+            while not validate_osd_utilization(osd_used=20):
+
+                # Create pvc
+                pvc_obj = pvc_factory()
+
+                # Create service_account to get privilege for deployment pods
+                sa_name = helpers.create_serviceaccount(
+                    pvc_obj.project.namespace
+                )
+
+                helpers.add_scc_policy(
+                    sa_name=sa_name.name, namespace=pvc_obj.project.namespace
+                )
+
+                pod_obj = helpers.create_pod(
+                    interface_type=constants.CEPHBLOCKPOOL,
+                    pvc_name=pvc_obj.name,
+                    namespace=pvc_obj.project.namespace,
+                    sa_name=sa_name.name,
+                    dc_deployment=True
+                )
+                helpers.wait_for_resource_state(
+                    resource=pod_obj, state=constants.STATUS_RUNNING
+                )
+                pod_obj.run_io(
+                    storage_type='fs', size='2G', runtime='60',
+                    fio_filename=f'{pod_obj.name}_io'
+                )
+
+        # Add capacity
+        osd_size = storage_cluster.get_osd_size()
+        count = storage_cluster.add_capacity(osd_size)
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        pod.wait_for_resource(
+            timeout=300,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=count * 3
+        )
+        assert ceph_health_check(), "Failed, New OSD pods failed to reach running state"
+
+        obj = CephCluster()
```

Comment:
I wouldn't name the variable `obj` but e.g. `cluster` but that's just a suggestion.